xonsh/completers/pip.py

```diff
@@ -22,7 +22,12 @@ def complete_pip(prefix, line, begidx, endidx, ctx):
     """
     Completes python's package manager pip
     """
-    if not PIP_RE.search(line):
+    case = [
+        not PIP_RE.search(line),
+        len(line.split()) > 3,
+        len(line.split()) > 2 and line.endswith(' '),
+    ]
```

Comment:
if you do something like this `line_len = len(line.split())` at the start of the function you will create only one list and calculate its length only once. a little more sufficient!